src/pymatching/sparse_blossom/search/search_graph.cc

```diff
@@ -134,7 +134,13 @@ pm::ImpliedWeight convert_rule(
         &nodes[i].neighbor_weights[nodes[i].index_of_neighbor(j == -1 ? nullptr : &nodes[j])];
     pm::weight_int* weight_pointer_j =
         j == -1 ? nullptr : &nodes[j].neighbor_weights[nodes[j].index_of_neighbor(&nodes[i])];
```

Comment:
Nit: would it be clearer to use `SIZE_MAX` instead of -1 here for consistency with the rest of the code base?